### PR TITLE
431 losses module results are not correct

### PR DESCRIPTION
**Pull request by sahand-asgarpour**
## Issue addressed
Solves #431

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [x] I HAVE discussed my solution with (other) members of the RA2CE team.

## What has been done?
Explain how you addressed the resolution of the related issue, what choices you made and why.

### Checklist
- [ ] Code is formatted using our custom `black` and `isort` definitions.
- [x] Tests are either added or updated.
- [x] Branch is up to date with `master`.
- [x] Updated documentation if needed.

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

**Comment by sahand-asgarpour** [comment]
When running test "tests.test_ra2ce_handler.TestRa2ceHandler.test_configure_handler_created_from_config_does_not_raise(Valid network config-Valid analysis config)" locally, I receive no error. But the test failed in TC.

File: ra2ce/.idea/ra2ce.iml
Comment (Carsopre) [inline ra2ce/.idea/ra2ce.iml:1]: Please remove this file from version control.

File: ra2ce/analysis/indirect/losses.py [hunk 1]
```diff
@@ -217,26 +217,23 @@ def _get_disrupted_criticality_analysis_results(
...
-            elif self.analysis.aggregate_wl == AggregateWlEnum.MIN:
-                self.criticality_analysis = criticality_analysis[
-                    criticality_analysis["EV1_min"] != 0
-                ]
-        else:
-            self.criticality_analysis = criticality_analysis
+        # filter out all links not affected by the hazard
+        if self.analysis.aggregate_wl == AggregateWlEnum.NONE:
+            self.criticality_analysis = criticality_analysis[
+                criticality_analysis["EV1_ma"] > float(self.analysis.threshold)
```
Comment (Carsopre) [inline ra2ce/analysis/indirect/losses.py:223]: `threshold` is already a `float`. I would not expect this casting to be needed. Please remove.

**Review by Carsopre — CHANGES_REQUESTED** [review]
Some remarks need to be processed, we can have a look afterwards to the failing test.

**Comment by sahand-asgarpour** [comment]
> Some remarks need to be processed, we can have a look afterwards to the failing test.

Remarks resolved.

File: ra2ce/analysis/indirect/losses.py [hunk 2]
```diff
@@ -548,6 +534,12 @@ def _populate_vehicle_loss_hour(
             raise Exception(
                 f"""{link_type_hazard_range} was not found in the introduced resilience_curve"""
             )
+        
+        if all(ratio <= 1 for ratio_tuple in relevant_curve["functionality_loss_ratio"] for ratio in ratio_tuple):
+                    divisor = 1
+        else:
+            divisor = 100
```
Comment (Carsopre) [inline ra2ce/analysis/indirect/losses.py:541]: For simple `if-else` statements like this you can simply do this to improve readability.
```suggestion
        divisor = 100
        if all(ratio <= 1 for ratio_tuple in relevant_curve["functionality_loss_ratio"] for ratio in ratio_tuple):
                    divisor = 1
```

File: ra2ce/analysis/indirect/losses.py [hunk 3]
```diff
@@ -445,11 +430,12 @@ def _create_result(vlh: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
                         performance_row[-1]["v"],
                         performance_key,
                     )
-                    if math.isnan(row_performance_change):
+                    if (math.isnan(row_performance_change) and row_connectivity == 0) or row_performance_change == 0:
                         self._calculate_production_loss_per_capita(
                             vehicle_loss_hours, vlh_row, event
                         )
-                    elif (u, v, k) == row_u_v_k:
+                    elif (not(math.isnan(row_performance_change) and math.isnan(row_connectivity)) and
```
Comment (Carsopre) [inline ra2ce/analysis/indirect/losses.py:437]: watch out when you start concatenating so many `if-else` statements with rather complex comparisons.

File: ra2ce/analysis/indirect/multi_link_redundancy.py [hunk 4]
```diff
@@ -40,34 +40,43 @@ def __init__(
         self.output_path = analysis_input.output_path
         self.hazard_names = analysis_input.hazard_names
 
-    def _update_time(self, gdf_calculated: pd.DataFrame, gdf_graph: gpd.GeoDataFrame):
+    def _update_time(self, gdf_calculated: pd.DataFrame, gdf_graph: gpd.GeoDataFrame) -> tuple:
```
Comment (Carsopre) [inline ra2ce/analysis/indirect/multi_link_redundancy.py:43]: `tuple` of what?

Comment (Carsopre) [reply ra2ce/analysis/indirect/multi_link_redundancy.py:43]: Perhaps `tuple[pd.Dataframe, gpd.GeoDataFrame]` ?

**Review by Carsopre — APPROVED** [review]
Minor remarks, you may merge afterwards :)

**Pull request merged**